docs: Explain how to add integration tests (#450)
XRPLF/xrpl-py: CONTRIBUTING.md

open index.html
```

## Write integration tests

> 1. Create a class that inherits `IntegrationTestCase` from `tests.integration.integration_test_case` to store all individual tests under (ex: `class TestWallet(IntegrationTestCase)`)

Reviewer: This should note that this is only necessary if adding functionality to a new part of the library

> 2. Create an async (likely) or sync (if specifically using an individual sync client) function that contains logic for the test

Reviewer 2: ```suggestion
2. Create an async function for each test case (unless the test is only being used for the sync client)
```
Parentheses pull the reader out of the flow of the sentence for a (hopefully quick) aside. It can be distracting to have them up front. Also, it's a bit easier to follow directions if the default path is clearly outlined first.

> 3. For customizability (on async functions only), include the `@test_async_and_sync` decorator to test it on sync/async JsonRpc/Websocket clients, manage import modules, specify only websockets, allow a set number of retries, or use the testnet (vs standalone rippled node). This decorator is used on a large majority of tests to ensure correctness on multiple clients (default is set to 4: AsyncWebsocket, Websocket, AsyncJsonRpc, JsonRpc). The few tests that do not utilize this decorator are usually specifically targeting coverage on an individual client. 

Reviewer: ```suggestion
3. For customizability (on async functions only), include the `@test_async_and_sync` decorator to test it on sync/async JsonRpc/Websocket clients, manage import modules, specify only websockets, allow a set number of retries, or use the testnet (vs standalone rippled node). This decorator is used on a large majority of tests to ensure correctness on multiple clients (default is set to 4: AsyncWebsocket, Websocket, AsyncJsonRpc, JsonRpc). The few tests that do not utilize this decorator are usually specifically targeting an individual client. 
```

> 3. For customizability (on async functions only), include the `@test_async_and_sync` decorator to test it on sync/async JsonRpc/Websocket clients, manage import modules, specify only websockets, allow a set number of retries, or use the testnet (vs standalone rippled node). This decorator is used on a large majority of tests to ensure correctness on multiple clients (by default, it tests all 4: `AsyncWebsocket`, `Websocket`, `AsyncJsonRpc`, `JsonRpc`). The few tests that do not utilize this decorator are usually specifically targeting coverage on an individual client.

Reviewer 2: ```suggestion
3. Include the `@test_async_and_sync` decorator to test against all client types, unless you specifically only want to test with one client. You can also use the decorator to:
       * Limit tests to sync/async only 
       * Limit the number of retries
       * Use Testnet instead of a standalone network
       * Manage import modules
```
* Highlighted the default action item up front (So you only need to read more if that's not what you're trying to do)
* Broke the list into bullets to make it easier to read
* Instead of describing the types of tests that don't use this decorator it's more clear to directly explain the reason they don't use it. (Ex. "If you only want to test one client, don't use this docorator" vs "Tests that only test one client usually don't have the decorator")

Also, what does "Manage import modules" mean?

Author: Manage import modules - in the test, we write everything asynchronously. The decorator handles converting everything async function to sync. As a result, you need to pass in the sync functions/modules to import inside the decorator.

> 4. Be sure to reuse pre-made values, `WALLET`, `DESTINATION`, `TESTNET_WALLET`, `TESTNET_DESTINATION`, `OFFER`, and `PAYMENT_CHANNEL`, from `tests/integrations/reusable_values.py`

Reviewer: just flagging that the testnet stuff will need to be removed with your other PR

> 5. Be sure to use condensed functions, like `submit_transaction` and `fund_wallet`, from `tests/integrations/it_utils.py`

Reviewer: note: `fund_wallet` should never be used inside an actual test (see current usage). Also, `submit_transaction_async` should always be used instead of `submit_transaction` (see current usage). The decorator does all the necessary replacement.

open _build/html/index.html
```

## Write integration tests

> 1. Create a .py file in correct subfolder of `tests/integrations` (ex: `test_wallet.py` in `tests/integrations/sugar`)

Reviewer: This step feels unnecessary

2. Create a class that inherits `IntegrationTestCase` to store all individual tests under (ex: `class TestWallet(IntegrationTestCase)`)

> 3. Create a function (can be sync or async) that contains logic for the test

Reviewer: The function has to be async if it's using the decorator

> 4. For customizability, include the `@test_async_and_sync` decorator on individual functions to test it on sync/async JsonRpc/Websocket clients, manage import modules, specify only websockets, allow a set number of retries, or use the testnet (vs standlone rippled node)

Reviewer: This should explain what the decorator is doing and why it's important to have on almost all of the tests (and what types of tests it shouldn't be on)

5. Be sure to reuse pre-made values, like `WALLET` or `TESTNET WALLET`, from `tests/integrations/reusable_values.py`

> 6. Be sure to use condensed functions, like `submit_transaction` and `fund_wallet`, from `tests/integrations/it_utils.py`

Reviewer: I'd list out all the reusable values here

open index.html
```

## Write integration tests
1. If adding functionality to a new part of the library, create new file with a class that inherits `IntegrationTestCase` from `tests.integration.integration_test_case` to store all individual tests under (ex: `class TestWallet(IntegrationTestCase)`). Otherwise, add to an existing file.
2. Create an async function for each test case (unless the test is only being used for the sync client)
3. Include the `@test_async_and_sync` decorator to test against all client types, unless you specifically only want to test with one client. You can also use the decorator to:
    * Limit tests to sync/async only 
    * Limit the number of retries
    * Use Testnet instead of a standalone network

>     * Manage import modules for sync equivalents of any async functions used 

Reviewer 2: Nit: Managed doesn't add much to the meaning
```suggestion
    * Import modules for sync equivalents of any async functions used 
```
